src/node/rpc/common_endpoint_registry.h

```diff
@@ -7,31 +7,171 @@
+  struct Quote
+  {
+    NodeId node_id = {};
+    std::string raw = {}; // < Hex-encoded
+
+    std::string error = {};
+    std::string mrenclave = {}; // < Hex-encoded
+  };
+
+  DECLARE_JSON_TYPE_WITH_OPTIONAL_FIELDS(Quote)
+  DECLARE_JSON_REQUIRED_FIELDS(Quote, node_id, raw)
+  DECLARE_JSON_OPTIONAL_FIELDS(Quote, error, mrenclave)
+
   /*
    * Extends the basic EndpointRegistry with methods which should be present
    * on all frontends
    */
   class CommonEndpointRegistry : public EndpointRegistry
   {
+  public:
+    AbstractNodeState& node;
+
+    ccf::TxStatus get_status_for_txid_v1(
+      kv::Consensus::View view, kv::Consensus::SeqNo seqno)
+    {
+      if (consensus != nullptr)
+      {
+        const auto tx_view = consensus->get_view(seqno);
+        const auto committed_seqno = consensus->get_committed_seqno();
+        const auto committed_view = consensus->get_view(committed_seqno);
+
+        return ccf::evaluate_tx_status(
+          view, seqno, tx_view, committed_view, committed_seqno);
+      }
+
+      return ccf::TxStatus::Unknown;
+    }
+
+    std::optional<std::pair<kv::Consensus::View, kv::Consensus::SeqNo>>
+    get_last_committed_txid_v1()
+    {
+      if (consensus != nullptr)
+      {
+        return consensus->get_committed_txid();
+      }
+
+      return std::nullopt;
+    }
+
+    nlohmann::json generate_openapi_document_v1(
+      kv::ReadOnlyTx& tx,
+      const std::string& title,
+      const std::string& description,
+      const std::string& document_version)
+    {
+      auto document =
+        ds::openapi::create_document(title, description, document_version);
+      build_api(document, tx);
+      return document;
+    }
+
+    std::optional<std::vector<uint8_t>> get_receipt_for_index_v1(
+      kv::Consensus::SeqNo seqno, std::string& error_reason)
+    {
+      try
+      {
+        if (history != nullptr)
+        {
+          try
+          {
+            return history->get_receipt(seqno);
+          }
+          catch (const std::exception& e)
+          {
+            error_reason = e.what();
+            return std::nullopt;
+          }
+        }
+
+        error_reason = "Node is not yet initialised";
+        return std::nullopt;
+      }
+      catch (const std::exception& e)
+      {
+        error_reason = "Exception thrown during execution";
+        return std::nullopt;
+      }
+    }
+
+    Quote get_quote_for_node_v1(kv::ReadOnlyTx& tx, NodeId node_id)
+    {
+      auto nodes_view = tx.get_read_only_view<ccf::Nodes>(Tables::NODES);
+      const auto node_info = nodes_view->get(node_id);
+      if (node_info.has_value())
+      {
+        Quote q;
+        q.node_id = node_id;
+
+        if (node_info->status == ccf::NodeStatus::TRUSTED)
+        {
+          q.raw = fmt::format("{:02x}", fmt::join(node_info->quote, ""));
+
+#ifdef GET_QUOTE
+          // TODO: Why don't we include this in BFT?
+          if (consensus != nullptr && consensus->type() != ConsensusType::BFT)
```

Comment:
@jumaffre @ashamis This was originally in `node_state.h` but moved here. Can you explain why the BFT path doesn't extract the code ID from the quote? We should add a comment if this is something we want to maintain long term.

Reply:
This was performance related. The BFT replays all the tx so every replica would attest every other replica. At one point attestation was very slow at times (not sure if this was the case) so the BFT tests were taking way too long

Reply:
That needs re-enabling now, good catch: #2064